sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreCoordinator.scala

```diff
@@ -122,13 +125,20 @@ private class StateStoreCoordinator(override val rpcEnv: RpcEnv)
     extends ThreadSafeRpcEndpoint with Logging {
   private val instances = new mutable.HashMap[StateStoreProviderId, ExecutorCacheTaskLocation]
 
-  override def receive: PartialFunction[Any, Unit] = {
-    case ReportActiveInstance(id, host, executorId) =>
+  override def receiveAndReply(context: RpcCallContext): PartialFunction[Any, Unit] = {
+    case ReportActiveInstance(id, host, executorId, otherProviderIds) =>
```

Comment:
Given the semantic of the request, I'm feeling more natural for this request to return provider ids which are assigned to this `ExecutorCacheTaskLocation`, say, "Here's the state store IDs you're registered as active.". That makes sure the request does the only a thing like we do for function - this change seems to do two different semantics of requests at once, and now return value doesn't look to be related to the origin semantic of request.

But I agree this change is better in point of performance-wise, as this doesn't require iterating on instances. Let's hear how others think about this.

Reply:
I'm OK to document ReportActiveInstance to describe the behavior. Probably changing the last parameter name to `providerIdsToCheckActive` (or `providerIdsToCheck` for shorter) would help understanding the behavior as well.

Reply:
Okay. I added a few comment and change the parameter name.